library/src/factory.cxx.in

```diff
@@ -49,7 +49,11 @@ reader* factory::getReader(const std::string& fileName)
   {
     for (auto r : p->getReaders())
```

Comment:
good occasion to rename this single letter variable. you could rename it `reader`